plugins/woocommerce/tests/php/src/Internal/DataStores/StockNotifications/StockNotificationsDataStoreTests.php

```diff
@@ -0,0 +1,535 @@
+<?php
+
+declare( strict_types = 1 );
+namespace Automattic\WooCommerce\Tests\Internal\DataStores\StockNotifications;
+
+use Automattic\WooCommerce\Internal\StockNotifications\Notification;
+use Automattic\WooCommerce\Internal\DataStores\StockNotifications\StockNotificationsDataStore;
+
+/**
+ * Class StockNotificationsDataStoreTests.
+ */
+class StockNotificationsDataStoreTests extends \WC_Unit_Test_Case {
+
+	/**
+	 * The data store instance.
+	 *
+	 * @var StockNotificationsDataStore
+	 */
+	private $data_store;
+
+	/**
+	 * Set up the test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		$this->data_store = wc_get_container()->get( StockNotificationsDataStore::class );
+	}
+
+	/**
+	 * Tear down the test.
+	 */
+	public function tearDown(): void {
+		parent::tearDown();
+		// Clean up all notifications.
+		global $wpdb;
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notifications" );
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notificationmeta" );
+		$wpdb->query( "TRUNCATE TABLE {$wpdb->prefix}wc_stock_notifications_logs" );
+	}
+
+	/**
+	 * Test that the stock notification data store is registered.
+	 */
+	public function test_stock_notification_data_store_is_registered() {
+		$store = new \WC_Data_Store( 'stock_notification' );
+		$this->assertTrue( is_callable( array( $store, 'read' ) ) );
+		$this->assertEquals( StockNotificationsDataStore::class, $store->get_current_class_name() );
+	}
+
+	/**
+	 * Test creating a notification with all properties.
+	 */
+	public function test_create_notification_with_all_properties() {
+		$notification = new Notification();
+
+		// Set all properties.
+		$notification->set_product_id( 1 );
+		$notification->set_user_id( 1 );
+		$notification->set_user_email( 'test@test.com' );
+		$notification->set_status( 'active' );
+		$notification->set_date_created( '2024-01-01 00:00:00' );
+		$notification->set_date_modified( '2024-01-01 00:00:00' );
+		$notification->set_date_subscribed( '2024-01-01 00:00:00' );
+		$notification->set_date_notified( '2024-01-01 00:00:00' );
```

Comment:
Maybe worth using different dates here to ensure they're not mixed up?

Reply:
Good call, added here https://github.com/woocommerce/woocommerce/pull/57062/commits/bf64d98486b08141396a0b9719ce58c1081b41d5